streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java

```diff
@@ -245,15 +245,28 @@ public void testTotalMetricDoesntDecrease() {
     }
 
     @Test
-    public void shouldGetStoreLevelTagMap() {
+    public void shouldGetStoreLevelTagMapForBuiltInMetricsLatestVersion() {
+        shouldGetStoreLevelTagMap(StreamsConfig.METRICS_LATEST);
+    }
+
+    @Test
+    public void shouldGetStoreLevelTagMapForBuiltInMetricsVersion0100To23() {
+        shouldGetStoreLevelTagMap(StreamsConfig.METRICS_0100_TO_23);
+    }
+
+    private void shouldGetStoreLevelTagMap(final String builtInMetricsVersion) {
         final String taskName = "test-task";
         final String storeType = "remote-window";
         final String storeName = "window-keeper";
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, THREAD_NAME, builtInMetricsVersion);
 
         final Map<String, String> tagMap = streamsMetrics.storeLevelTagMap(taskName, storeType, storeName);
 
         assertThat(tagMap.size(), equalTo(3));
-        assertThat(tagMap.get(StreamsMetricsImpl.THREAD_ID_TAG_0100_TO_23), equalTo(THREAD_NAME));
+        assertThat(
+            tagMap.get(builtInMetricsVersion.equals(StreamsConfig.METRICS_LATEST) ? StreamsMetricsImpl.THREAD_ID_TAG
+                : StreamsMetricsImpl.THREAD_ID_TAG_0100_TO_23),
```

Comment:
Maybe get the value of this expression on the line above. IMHO it will make it easier to see what the `assertThat` is doing.

Reply:
Done